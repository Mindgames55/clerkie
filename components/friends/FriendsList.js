'use client'

import { useState } from 'react';
import styles from './FriendsList.module.css'
import Friend from './Friend';

// TODO use real data
const friendMock = [{
    first_name: 'Jhon',
    last_name: 'Smith',
    email: 'email@gmail.com',
    'phone number': '(555) 456 4567',
    status: 'close',
    id: '1'
}]

export default function FriendsList({children}) {
    const [ isInitialLoad, setInitialLoad ] = useState(true);
    const friends = friendMock
    // TODO figure out a way to avoid using magic 80px
    return (
        <ul className={styles.container}>
            {isInitialLoad && children}
            {friends.map(friend => <Friend friend={friend} key={friend.id}/>)}
        </ul>
    )
}