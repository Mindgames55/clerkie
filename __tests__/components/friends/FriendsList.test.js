import { render } from '@testing-library/react';
import FriendsList from '@/components/friends/FriendsList';

jest.mock('../../../components/hooks/useIntersectionObserver.js')
jest.mock('../../../components/hooks/useFriends.js', () => () => ({
    friends: [],
    loading: false,
    getFriends: jest.fn()
}))
it('renders FriendsList correctly', () => {
    const { container } = render(<FriendsList />);
    expect(container).toMatchSnapshot();
});