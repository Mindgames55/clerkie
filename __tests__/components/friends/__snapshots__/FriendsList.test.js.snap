// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FriendsList correctly 1`] = `
<div>
  <ul
    class="container"
  >
    <li
      class="container"
    >
      <span
        class="name"
      >
        Jhon Smith
      </span>
      <span
        class="close"
      >
        Close Friends
      </span>
      <span
        class="metadata"
      >
        <span
          class="email"
        >
          email@gmail.com
        </span>
        <span />
      </span>
    </li>
  </ul>
</div>
`;
